File: src/jvm/desig/pattern/chainofresponsibility/AbstractHandler.java
package jvm.desig.pattern.chainofresponsibility;

/**
 * 责任链接口
 */
public abstract class AbstractHandler {
    AbstractHandler abstractHandler;

    void setSuccessor(AbstractHandler abstractHandler) {
        this.abstractHandler = abstractHandler;
    }

    public abstract void request(int requestNumber);
}

File: src/kt/desig/pattern/chainofresponsibility/AbstractHandler.kt
package kt.desig.pattern.chainofresponsibility

/**
 * 责任链接口
 */
abstract class AbstractHandler {
    lateinit var abstractHandler: AbstractHandler
    internal fun setSuccessor(abstractHandler: AbstractHandler) {
        this.abstractHandler = abstractHandler
    }
    abstract fun request(requestNumber: Int)
}